x/concentrated-liquidity/lp.go

```diff
@@ -417,22 +420,20 @@ func (k Keeper) addToPosition(ctx sdk.Context, owner sdk.AccAddress, positionId
 		return 0, osmomath.Int{}, osmomath.Int{}, err
 	}
 
-	anyPositionsRemainingInPool, err := k.HasAnyPositionForPool(ctx, position.PoolId)
+	pool, err := k.GetConcentratedPoolById(ctx, position.PoolId)
 	if err != nil {
 		return 0, osmomath.Int{}, osmomath.Int{}, err
 	}
 
+	anyPositionsRemainingInPool := k.PoolHasPosition(ctx, pool)
```

Comment:
Does this also give us speed up compared to calling `HasAnyPositionForPool` directly?

Reply:
Yes! `HasAnyPositionForPool` runs an iterator.  PoolHasPosition just checks data off the pool struct directly

Reply:
If so, isn't it also possible to change `HasAnyPositionForPool` to do `GetConcentratedPoolById` then call `PoolHasPosition` instead of running an iterator?

Reply:
yeah, it was mostly that we still need the old method as well for withdraw position